src/auth/guards/gpl-auth.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Reflector } from '@nestjs/core';
+import { AuthGuard } from '@nestjs/passport';
+import { ExecutionContext } from '@nestjs/common';
+import { GqlExecutionContext } from '@nestjs/graphql';
+
+import { whiteListResolver } from '../auth.constants';
+
+export class GqlAuthGuard extends AuthGuard('jwt') {
+  constructor(private reflector: Reflector) {
+    super();
+  }
+
+  canActivate(context: ExecutionContext) {
+    const ctx = GqlExecutionContext.create(context);
+    const fieldName = ctx.getInfo().fieldName;
+    if (whiteListResolver.includes(fieldName)) return true;
```

Comment:
Have you tried to use the `@UseGuards` decorator to secure queries/mutations?

Ref: https://github.com/nestjs/nest/blob/master/sample/19-auth-jwt/src/app.controller.ts#L10

Reply:
yes , i do, but i made globally instead of making in each resolver method, I set it in the main module